mocha-shared-behaviors/index.md

```diff
@@ -0,0 +1,494 @@
+
+```javascript
+/// helpers.js
+export const expectUserLike = user => ({
+  toHaveNameFirstAs: expectation => {
+    expect(user.name.first).to.equal(expectation);
+  },
+  toHaveNameLastAs: expectation => {
+    expect(user.name.last).to.equal(expectation);
+  },
+  toHaveFullnameThatReturnAs: expectation => {
+    expect(user.fullname()).to.equal(expectation);
+  }
+});
+
+/// user.test.js
+let user = 'foo';
+const constructorArgs = ['tobi', 'holowaychuk'];
+
+describe('User', () => {
+  beforeEach(() => {
+    user = new User(...constructorArgs);
+  });
+
+  it('should have .name.first', () => {
+    expectUserLike(user).toHaveNameFirstAs(constructorArgs[0]);
+  });
+
+  // other tests
+});
+```
+
+Now, this shared behavior isn't isolated anymore. And it's simple :kiss:!
+
+Not being able test every aspect of the behavior, or define an order, spec description, setup and tear down, could be an important downside for some use cases. Yet, in my opinion, this isn't really needed as often as you may think.
+
+This approach is often my preference. It's simple, explicit **and** permits definition of shared behaviors in separate files.
+
+Yet, I only use it if separate files is an absolute requirement.
+
+## The power of closures <a name="closures-only"></a>
+
+> complete example on Github :arrow_right: [test/closure](https://github.com/noelmace/mocha-shared-behaviors/tree/master/test/closure)
+
+If it isn't, simply use the lambda closure to share data between your shared behaviors.
+
+Take the first example, from the Mocha Wiki. `user.test.js` and `admin.test.js` are actually in a single file, `test.js`. `User` and `Admin` are from the same "feature scope," so it feels right and logical to test those two as one.
+
+With this idea, let's refactor a little.
+
+```javascript
+let userLike;
+
+const shouldBehaveLikeAUser = (firstName, lastName) => {
+  it('should have .name.first', () => {
+    expect(userLike.name.first).to.equal(firstName);
+  });
+  // other tests
+};
+
+describe('User', () => {
+  const firstName = 'tobi';
+  const lastName = 'holowachuk';
+
+  beforeEach(() => {
+    userLike = new User(firstName, lastName);
+  });
+
+  shouldBehaveLikeAUser(firstName, lastName);
+});
+
+describe('Admin', () => {
+  const firstName = 'foo';
+  const lastName = 'bar';
+
+  beforeEach(() => {
+    userLike = new Admin(firstName, lastName);
+  });
+
+  shouldBehaveLikeAUser(firstName, lastName);
+
+  it('should be an .admin', () => {
+    expect(userLike.admin).to.be.true;
+  });
+});
+```
+
+#### my 2 cents about this method
+
+Most of the time, I write my tests in a [Given-When-Then](https://martinfowler.com/bliki/GivenWhenThen.html) style. This is more a way to think about _how_ you test than a way to write tests! Here is an example to clarify.
+
+> **FYI** Here, I only "replaced" `it` and `context` by `then.it`, `given` & `when` in order to make it more simple to understand.
```

Comment:
for me this whole paragraph seems a little out of context... => it even states `I DO NOT advise you to do this in your project!`
so as a reader I am wondering why did I read it?

overall it goes very smoothly from top to bottom now 💪 
good advice and step by step small improvements... this seems a little too "different" too "extra"... with not enough information

I would leave it out => if you want to teach this it should have it's own blog post with extra info how you ended up there and how you can use it for real